packages/build/tests/plugins/tests.js

```diff
@@ -1064,6 +1064,13 @@ test('Provided --node-path version produces a node version warning if the build
   await runFixture(t, 'engines', { flags: { nodePath, mode: 'buildbot' } })
 })
 
+test('Provided --node-path version has no effect for plugin executions where the feature flag plugins_system_node_version is set', async (t) => {
+  const nodePath = getNodePath('9.0.0')
+  await runFixture(t, 'engines', {
+    flags: { nodePath, mode: 'buildbot', featureFlags: { buildbot_build_plugins_system_node_version: true } },
```

Comment:
Instead of creating a test with the feature flag, should we set the feature flag on all tests in https://github.com/netlify/build/blob/251fe66efa275d1212ff1c7653785dbf10ce5f89/packages/build/tests/helpers/main.js#L44?

One downside of that approach is that the tests will now reflect the future/feature flagged logic instead of the current logic.
But an upside is that it allows running all the tests with the future behavior. This gives us some confidence when increasing the LaunchDarkly feature flag progressively to 100%.

Reply:
I'm a bit reluctant tbh.
- We still want to have test cases where the flag isn't set. We can always disable it manually on each test but I wonder if meanwhile we might introduce something that ends up breaking builds for a large chunk of users without us realising  (essentially the other side of the coin as you mentioned)?
- Setting it by default will change the snapshots for all of the tests I believe(?), making this review process quite cumbersome and probably error prone? 🤔

If we really want to cover all the bases, maybe for the time being we could run **all** the plugin tests with both feature flag set to `true` and `false`? What do you think?

Reply:
For information, we discussed it on Slack and decided to try running all the tests with the feature flag.